src/bootstrap/src/core/download.rs

```diff
@@ -742,6 +745,38 @@ download-rustc = false
         let llvm_root = self.ci_llvm_root();
         self.unpack(&tarball, &llvm_root, "rust-dev");
     }
+
+    pub(crate) fn maybe_download_gccjit(&self) {
+        if !self.gcc_download_gccjit {
+            return;
+        }
+        self.download_gccjit(&detect_gccjit_sha());
+    }
+
+    fn download_gccjit(&self, gcc_sha: &str) {
+        let help_on_error = "ERROR: failed to download libgccjit
+
+    HELP: There could be two reasons behind this:
+        1) The host triple is not supported for `download-gccjit`.
+        2) Old builds get deleted after a certain time.
+    ";
+        if !self.build.triple.contains("linux") || !self.build.triple.contains("x86_64") {
+            eprintln!("{help_on_error}");
+            return;
+        }
+        let rustc_cache = self.libgccjit_folder(gcc_sha);
+        if !rustc_cache.exists() {
+            t!(fs::create_dir_all(&rustc_cache));
+        }
+
+        let lib_path = rustc_cache.join("libgccjit.so.0");
+        if !lib_path.exists() {
+            let url = format!(
+                "https://github.com/rust-lang/gcc/releases/download/master-{gcc_sha}/libgccjit.so",
```

Comment:
I picked this one because I don't know if relying on github CI artifacts is better or worse. If someone from the infra team has guidance there or an opinion, it'd be very helpful.

Reply:
I would expect that this would be implemented in the same way we implemented LLVM: a component shipped with every commit built on master of rust-lang/rust and a download of that component. This looks like it's using a pretty different approach, I would prefer to keep things consistent if we can.